posts/ds_stack_setup.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Fitting IPython Notebooks, Spark and Cassandra all together
+author: carlos
+date: "2015-10-23"
+description: An article on how to configure all these tools together
+tags: ["Cassandra", "Spark", "DataStax Enterprise", "IPython Notebooks", "IPython"]
+categories: ["devops"]
+---
+
+Yesterday after more than a whole week working on this I've finally managed to set all this stack up.
+
+This stack is one of the most hot and trending topics nowadays as it is very useful for BigData, specifically for data exploration purposes.
+
+My starting point was a 3 nodes Cassandra cluster, intended for analytics, data exploration and adhoc reporting. This cluster was running
+DSE (DataStax Enterprise) 4.6, which ships with Cassandra 2.0. Worths mentioning that the cluster was running in Cassandra only mode.
+
+After having attended to the latest [Cassandra Summit '15](http://mrcalonso.com/notes-on-cassandra-summit-2015/), it was made clear not to use
+Spark with any Cassandra earlier than 2.1.8, because the integration was buggy. Therefore:
+
+1. Upgrade DSE to latest (4.8) version, that includes Cassandra 2.1.10.
+  * Step by step instructions here: [http://docs.datastax.com/en/upgrade/doc/upgrade/datastax_enterprise/upgradeDSE47.html](http://docs.datastax.com/en/upgrade/doc/upgrade/datastax_enterprise/upgradeDSE47.html)
+2. Next step is to enable Spark on the cluster. This is one of the points where relying on something like DSE comes really helpful, as the DSE distribution comes
+with a Spark installation and the Cassandra-Spark connector already configured and optimised for maximum compatbility and throughput. It is also really easy to
+enable Spark on the nodes.
+  * For more information on this, head here: [http://docs.datastax.com/en/datastax_enterprise/4.8/datastax_enterprise/anaHome/anaHomeTOC.html](http://docs.datastax.com/en/datastax_enterprise/4.8/datastax_enterprise/anaHome/anaHomeTOC.html)
+3. Once I had Cassandra and Spark running on all nodes and one of them was designated as the Spark Master (first of the Cassandra seeds by default, but you can check it with `dse client-tool spark master-address`), it's time to install IPython and all its dependencies:
+  1. First step is to install all system required packages (use apt-get, yum, etc depending on your OS): `sudo apt-get install build-essential libcurl4-openssl-dev libssl-dev zlib1g-dev libpcre3-dev gfortran libblas-dev libblas3gf liblapack3gf liblapack-dev libncurses5-dev libatlas-dev libatlas-base-dev libscalapack-mpi1 libscalapack-pvm1 liblcms-utils python-imaging-doc python-imaging-dbg libamd2.3.1 libjpeg-turbo8 libjpeg8 liblcms1 libumfpack5.6.2 python-imaging libpng12-0 libpng12-dev libfreetype6 libfreetype6-dev libcurl4-gnutls-dev python-pycurl-dbg git-core cython libhdf5-7 libhdf5-serial-dev python-egenix-mxdatetime vim python-numpy python-scipy pandoc openjdk-7-jdk`
```

Comment:
That's just Markdown. It'll be displayed as roman numbers indented below (I think)